lib/ansiblelint/__init__.py

```diff
@@ -196,7 +206,9 @@ def listtags(self):
     @classmethod
     def create_from_directory(cls, rulesdir):
         result = cls()
```

Comment:
After your change of passing rulesdir into the initializer, it's not really needed anymore:
```suggestion
        return cls(rulesdir)
```
which probably means that any invocations of `RulesCollection.create_from_directory(rulesdir)` should be replaced with `RulesCollection(rulesdir)` and this classmethod should be deleted. Or maybe you should implement a public method that populates `rules` and call it here as
```suggestion
        obj = cls()
        obj.populate_rule_dirs(rulesdirs)
        return obj
```
And then there'd be no need to rewrite the initializer. (But I think that the initializer rewrite seems to be a more favorable approach)

Reply:
I agree the create_from_directory can probably go. I removed it from __main__.py and the only other places were a bunch of tests. I wasn't sure if I should go down that route in this PR. I will go ahead an do that now.